File: LeetCode/Easy/1539-kth-missing-positive-number/1539-kth-missing-positive-number.go
func findKthPositive(arr []int, k int) int {
    i := 0;
    min_num := 1;
    for j := 0; j < len(arr); j++ {
        if arr[j] != min_num {
            if arr[j] - min_num >= k - i { return min_num + k - i - 1; }
            i += arr[j] - min_num;
            min_num = arr[j] + 1;
            continue
        }
        min_num += 1
    }
    return min_num + k - i - 1;
}

File: LeetCode/Easy/1539-kth-missing-positive-number/1539-kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int i = 0;
        int min_num = 1;
        for (int v : arr){
            if (v != min_num){
                if (v - min_num >= k - i) return min_num + k - i - 1;
                else {
                    i += v - min_num;
                    min_num = v + 1;
                }
            }
            else min_num += 1;
        }
        return min_num + k - i - 1;
    }
}